core/app/models/spree/line_item.rb

```diff
@@ -138,49 +114,68 @@ def options=(options={})
       opts = options.dup # we will be deleting from the hash, so leave the caller's copy intact
 
       currency = opts.delete(:currency) || order.try(:currency)
```

Comment:
I don't think you are right. Since `currency=` is an attribute setter method defined by AR, it's getting set instead of a new local variable is used. That's why `self.currency = ` is redundant usage of `self` ;) But, I can be wrong.